circuit-benchmarks/src/pi_circuit.rs

```diff
@@ -0,0 +1,113 @@
+//! State circuit benchmarks
+#[cfg(test)]
+mod tests {
+    use ark_std::{end_timer, start_timer};
+    use eth_types::Word;
+    use halo2_proofs::plonk::{create_proof, keygen_pk, keygen_vk, verify_proof};
+    use halo2_proofs::poly::kzg::commitment::{KZGCommitmentScheme, ParamsKZG, ParamsVerifierKZG};
+    use halo2_proofs::poly::kzg::multiopen::{ProverSHPLONK, VerifierSHPLONK};
+    use halo2_proofs::poly::kzg::strategy::SingleStrategy;
+    use halo2_proofs::{
+        halo2curves::bn256::{Bn256, Fr, G1Affine},
+        poly::commitment::ParamsProver,
+        transcript::{
+            Blake2bRead, Blake2bWrite, Challenge255, TranscriptReadBuffer, TranscriptWriterBuffer,
+        },
+    };
+    use rand::SeedableRng;
+    use rand_chacha::ChaCha20Rng;
+    use rand_xorshift::XorShiftRng;
+    use std::env::var;
+    use zkevm_circuits::pi_circuit::{PiCircuit, PublicData};
+    use zkevm_circuits::test_util::rand_tx;
+
+    #[cfg_attr(not(feature = "benches"), ignore)]
+    #[test]
+    fn bench_pi_circuit_prover() {
+        let degree: u32 = var("DEGREE")
+            .unwrap_or("15".to_string())
+            .parse()
+            .expect("Cannot parse DEGREE env var as u32");
+
+        const MAX_TXS: usize = 10;
+        const MAX_CALLDATA: usize = 128;
```

Comment:
The full circuit will be filled in as much as possible for any circuit height even with these fixed values?

Reply:
For TX yes, it fills MAX_TXS txs, but not enough calldata bytes as rand_tx() simply fills just a few bytes. I thought it might be ok for calldata contents, anyway, we could also update the rand_tx to support this.

Reply:
Makes sense, seems fine to me.